akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/Running.scala

```diff
@@ -97,6 +109,48 @@ private[akka] object Running {
   import InternalProtocol._
   import Running.RunningState
 
+  import scala.concurrent.duration._
+  import akka.actor.typed.scaladsl.AskPattern._
+
+  private val log = setup.log
+
+  def startReplicationStream(state: RunningState[S]): Unit = {
+    setup.activeActive.map { aa =>
+      val query = PersistenceQuery(setup.context.system)
+      aa.allReplicas.zipWithIndex.foreach {
+        case (replica, _) =>
+          if (replica != aa.replicaId) {
+            val seqNr = state.seenPerReplica(replica)
+            val pid = PersistenceId.replicated(aa.aaContext.id, replica)
+            // FIXME use index to get plugin id to allow diff databases per replica
+            // and config section to define the read journal config
+            // https://github.com/akka/akka/issues/29257
+            val replication = query.readJournalFor[EventsByPersistenceIdQuery](LeveldbReadJournal.Identifier)
+
+            implicit val timeut = Timeout(30.seconds)
+            implicit val scheduler = setup.context.system.scheduler
+
+            val source = RestartSource.withBackoff(2.seconds, 10.seconds, randomFactor = 0.2) { () =>
+              replication
+                .eventsByPersistenceId(pid.id, seqNr + 1, Long.MaxValue)
+                .mapAsync(1) { eventEnvelope: EventEnvelope =>
+                  setup.context.self.ask[ReplicatedEventAck.type] { replyTo =>
+                    log.debug(s"{} received replicated event from {}: {}", aa.replicaId, replica, eventEnvelope)
+                    ReplicatedEventEnvelope(eventEnvelope.event.asInstanceOf[ReplicatedEvent[E]], replyTo)
```

Comment:
I guess we want the ask to get back pressure, but doesn't that make it quite unfair vs local commands/events? (also pricey with a ask-actor + timeout for every event?)

Reply:
It does save it to the DB so i think we need the back pressure.

Alternatively, we could maybe use reliable delivery somehow?